packages/client/hmi-client/src/components/models/Model.vue

```diff
@@ -125,7 +184,64 @@ const props = defineProps<ModelProps>();
 
 const relatedTerariumArtifacts = ref<ResultType[]>([]);
 const model = ref<ITypedModel<PetriNet> | null>(null);
-const isEditing = ref(false);
+const isEditing = ref<boolean>(false);
+
+const equation = ref<string>('');
+const selectedRow = ref();
+const height = ref(400);
+
+let draggableY = 0;
+
+const mathmode = ref('mathJAX');
+
+// Test equation.  Was thinking this would probably eventually live in model.mathLatex or model.mathML?
+const modelMath = String.raw`\frac{dS}{dt} = -\beta IS \frac{dI}{dt} = \
+ 	\beta IS - \gamma I \frac{dR}{dt} = \gamma I`;
+
+// Another experiment using a map to automatically select highlighted version of the latex formula
+// this would require the backend service to provide a map of the eq.  Might be a little challenging.
+// const equationMap = {
+// 	default: String.raw`\frac{dS}{dt} = -\beta IS \frac{dI}{dt} = \
+// 	\beta IS - \gamma I \frac{dR}{dt} = \gamma I`,
+// 	S: String.raw`\frac{d\colorbox{red}{S}}{dt} = -\beta I{\color{red}{S}} \frac{dI}{dt} = \
+// 	\beta I{\color{red}{S}} - \gamma I \frac{dR}{dt} = \gamma I`,
+// 	I: String.raw`\frac{dS}{dt} = -\beta {\color{red}I}S \frac{dI}{dt} = \
+// 	\beta {\color{red}I}S - \gamma {\color{red}I} \frac{dR}{dt} = \gamma {\color{red}I}`,
+// 	R: String.raw`\frac{dS}{dt} = -\beta IS \frac{dI}{dt} = \
+// 	\beta IS - \gamma I \frac{d\color{red}{R}}{dt} = \gamma I`
+// };
+
+// DataTable click handler for State Variables.  Currently used to do the highlighting.
+const onRowClick = () => {
+	if (selectedRow.value) {
+		equation.value = modelMath.replaceAll(
+			selectedRow.value.sname,
+			String.raw`{\color{red}${selectedRow.value.sname}}`
+		);
+	} else {
+		equation.value = modelMath;
+	}
+};
+
+const updateFormula = (formula_string: string) => {
+	equation.value = formula_string;
+};
+
+const resize = (event: MouseEvent) => {
+	height.value += event.clientY - draggableY;
+	draggableY = event.clientY;
+};
+
+const stopResize = () => {
+	document.removeEventListener('mousemove', resize);
+	document.removeEventListener('mouseup', stopResize);
+};
+
+const startResize = (event: MouseEvent) => {
+	draggableY = event.clientY;
+	document.addEventListener('mousemove', resize);
+	document.addEventListener('mouseup', stopResize);
+};
```

Comment:
Because of all of this, I wonder if a new component `tera-grid` or something like that would be better for reusability across the app.